src/Storage/Storage/DatalakeGen2/Cmdlet/GetAzDataLakeGen2FileContent.cs

```diff
@@ -0,0 +1,264 @@
+        public GetAzDataLakeGen2ItemContentCommand(IStorageBlobManagement channel)
+        {
+            Channel = channel;
+            fileNameResolver = new BlobToFileSystemNameResolver(() => NameUtil.WindowsMaxFileNameLength);
+        }
+
+        /// <summary>
+        /// Download blob to local file
+        /// </summary>
+        /// <param name="blob">Source blob object</param>
+        /// <param name="filePath">Destination file path</param>
+        internal virtual async Task DownloadBlob(long taskId, IStorageBlobManagement localChannel, CloudBlob blob, string filePath)
+        {
+            string activity = String.Format(Resources.ReceiveAzureBlobActivity, blob.Name, filePath);
+            string status = Resources.PrepareDownloadingBlob;
+            ProgressRecord pr = new ProgressRecord(OutputStream.GetProgressId(taskId), activity, status);
+            DataMovementUserData data = new DataMovementUserData()
+            {
+                Data = blob,
+                TaskId = taskId,
+                Channel = localChannel,
+                Record = pr,
+                TotalSize = blob.Properties.Length
+            };
+
+            await DataMovementTransferHelper.DoTransfer(() =>
+                {
+                    return this.TransferManager.DownloadAsync(blob, filePath,
+                        new DownloadOptions()
+                        {
+                            DisableContentMD5Validation = !this.checkMd5
+                        },
+                        this.GetTransferContext(data),
+                        this.CmdletCancellationToken);
+                },
+                data.Record,
+                this.OutputStream).ConfigureAwait(false);
+
+            //this.WriteCloudBlobObject(data.TaskId, data.Channel, blob);
+            WriteDataLakeGen2Item(localChannel, (CloudBlockBlob)blob, taskId: data.TaskId);
+        }
+
+        /// <summary>
+        /// get blob content
+        /// </summary>
+        /// <param name="blob">source CloudBlob object</param>
+        /// <param name="fileName">destination file path</param>
+        /// <param name="isValidBlob">whether the source FileSystem validated</param>
+        /// <returns>the downloaded blob object</returns>
+        internal void GetBlobContent(CloudBlob blob, string fileName, bool isValidBlob = false)
+        {
+            if (null == blob)
+            {
+                throw new ArgumentNullException(typeof(CloudBlob).Name, String.Format(Resources.ObjectCannotBeNull, typeof(CloudBlob).Name));
+            }
+
+            ValidateBlobType(blob);
+
+            string filePath = GetFullReceiveFilePath(fileName, blob.Name);
+
+            if (!isValidBlob)
+            {
+                ValidatePipelineCloudBlob(blob);
+            }
+
+            //create the destination directory if not exists.
+            String dirPath = System.IO.Path.GetDirectoryName(filePath);
+
+            if (!Directory.Exists(dirPath))
```

Comment:
It will be false if dirPath is a file. Am I right?

Reply:
Yes, if dirPath is a file, it will be false.

Reply:
Then `Directory.CreateDirectory(dirPath)` will raise an exception and told users that there is already a file at `dirPath`. Sounds make sense.